server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/ClusterBalanceStats.java

```diff
@@ -269,7 +275,8 @@ public static NodeBalanceStats readFrom(StreamInput in) throws IOException {
                 in.getTransportVersion().onOrAfter(TransportVersions.V_8_12_0) ? in.readVInt() : -1,
                 in.readDouble(),
                 in.readLong(),
-                in.readLong()
+                in.readLong(),
+                in.readDouble()
```

Comment:
We need guard the read and write of the new field with a new transport version similar to what is done a few lines above for `TransportVersions.V_8_12_0`.